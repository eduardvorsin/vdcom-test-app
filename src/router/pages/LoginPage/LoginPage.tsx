import React from 'react';
import { Box, Typography, Container, styled } from '@mui/material';
import Login from '../../../components/UI/Login/Login';
import Logo from '../../../components/UI/Logo/Logo';

const LoginWrapper = styled(Box)(({ theme }) => ({
  height: '100vh',
  margin: '0 auto',
  maxWidth: '500px',
  minHeight: '505px',
  display: 'flex',
  justifyContent: 'center',
  flexDirection: 'column',

  [theme.breakpoints.up('sm')]: {
    minHeight: '585px',
  },
  [theme.breakpoints.up('md')]: {
    minHeight: '692px',
    maxWidth: '654px',
  },
  [theme.breakpoints.up('lg')]: {
    minHeight: '790px',
  },
}));

const LoginLogo = styled(Logo)(({ theme }) => ({
  alignSelf: 'center',
  marginBottom: '28px',

  [theme.breakpoints.up('sm')]: {
    marginBottom: '68px',
  },
  [theme.breakpoints.up('md')]: {
    marginBottom: '108px',
  },
  [theme.breakpoints.up('lg')]: {
    marginBottom: '148px',
  },
}));

const LoginTitle = styled(Typography)(({ theme }) => ({
  lineHeight: '1.46',
  marginBottom: '20px',
  textAlign: 'center',

  [theme.breakpoints.up('sm')]: {
    marginBottom: '25px',
  },
  [theme.breakpoints.up('md')]: {
    marginBottom: '30px',
  },
  [theme.breakpoints.up('lg')]: {
    marginBottom: '35px',
  },
}));

const LoginPage = () => (
  <Container
    maxWidth='xl'
    data-testid='login-page'
  >
    <LoginWrapper>
      <LoginLogo />

      <LoginTitle
        variant='h3'
        component='h1'
      >
        Welcome To CRM System Sign In To Your Account
      </LoginTitle>

      <Login />
    </LoginWrapper>
  </Container>

);

export default LoginPage;
