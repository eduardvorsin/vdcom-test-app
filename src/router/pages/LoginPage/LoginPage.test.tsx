import React from 'react';

import { act, render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import LoginPage from './LoginPage';
import addMemoryRouter from '../../../tests/helpers/addMemoryRouter';
import wrapWithRedux from '../../../tests/helpers/wrapWithRedux';

const routes = [{
  path: '/',
  element: <LoginPage />,
}];

const router = createMemoryRouter(routes, {
  initialEntries: ['/'],
});

describe('LoginPage tests', () => {
  it('renders correctly', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByRole<HTMLTitleElement>('heading', { name: /Welcome To CRM System Sign In To Your Account/i })).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByTestId<HTMLDivElement>('login-page')).toMatchSnapshot();
  });
});

describe('LoginPage integration tests', () => {
  it('should show an error if the login and password field is not filled in and the sign in button was pressed', async () => {
    const user = userEvent.setup();
    render(wrapWithRedux(addMemoryRouter({ initialEntries: ['/login'] }), {}));

    await act(async () => {
      await user.type(screen.getByLabelText<HTMLInputElement>(/login/i), 'rrooot2');
      await user.type(screen.getByLabelText<HTMLInputElement>(/password/i), 'rrooot2');
    });

    user.click(screen.getByRole<HTMLButtonElement>('button', { name: /Sign in/i }));

    expect(await screen.findByText<HTMLHeadingElement>(/Error/i)).toBeInTheDocument();
    expect(await screen.findByText<HTMLDivElement>(/not found/i)).toBeInTheDocument();
  });

  it('is a snapshot with error message', async () => {
    const user = userEvent.setup();
    render(wrapWithRedux(addMemoryRouter({ initialEntries: ['/login'] }), {}));

    await act(async () => {
      await user.type(screen.getByLabelText<HTMLInputElement>(/login/i), 'rrooot2');
      await user.type(screen.getByLabelText<HTMLInputElement>(/password/i), 'rrooot2');
    });

    user.click(screen.getByRole<HTMLButtonElement>('button', { name: /Sign in/i }));

    await screen.findByText<HTMLHeadingElement>(/Error/i);
    expect(screen.getByTestId<HTMLDivElement>('login-page')).toMatchSnapshot();
  });
});
