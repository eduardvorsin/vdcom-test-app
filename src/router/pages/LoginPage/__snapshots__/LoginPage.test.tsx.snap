// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage integration tests is a snapshot with error message 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  data-testid="login-page"
>
  <div
    class="MuiBox-root css-sgip5j"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cg55gr-MuiTypography-root-MuiLink-root"
      data-testid="logo-link"
      href="/contacts"
    >
      <img
        alt="logo"
        class="css-12pywbb"
        src="assets"
      />
    </a>
    <h1
      class="MuiTypography-root MuiTypography-h3 css-16d7d20-MuiTypography-root"
    >
      Welcome To CRM System Sign In To Your Account
    </h1>
    <div
      class="MuiBox-root css-0"
      data-testid="login"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-14a4mxl-MuiPaper-root-MuiAlert-root"
        role="alert"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
          >
            Error
          </div>
          The user with the same name and password was not found, try again
        </div>
      </div>
      <form
        action="/login"
        data-testid="login-form"
        method="post"
        novalidate=""
      >
        <div
          class="MuiBox-root css-utmfdt"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17c6y5f-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r6:"
                id=":r6:-label"
              >
                Login
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="username"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r6:"
                  name="username"
                  required=""
                  type="text"
                  value="rrooot2"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      Login
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17c6y5f-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r7:"
                id=":r7:-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r7:"
                  name="password"
                  required=""
                  type="password"
                  value="rrooot2"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      Password
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-tddfjc"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-kcn0n4-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Sign in
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              >
                <span
                  class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                  style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
                >
                  <span
                    class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
                  />
                </span>
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`LoginPage tests is a basic snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  data-testid="login-page"
>
  <div
    class="MuiBox-root css-sgip5j"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cg55gr-MuiTypography-root-MuiLink-root"
      data-testid="logo-link"
      href="/contacts"
    >
      <img
        alt="logo"
        class="css-12pywbb"
        src="assets"
      />
    </a>
    <h1
      class="MuiTypography-root MuiTypography-h3 css-16d7d20-MuiTypography-root"
    >
      Welcome To CRM System Sign In To Your Account
    </h1>
    <div
      class="MuiBox-root css-0"
      data-testid="login"
    >
      <form
        action="/login"
        data-testid="login-form"
        method="post"
        novalidate=""
      >
        <div
          class="MuiBox-root css-utmfdt"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17c6y5f-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Login
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="username"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Login
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17c6y5f-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r3:"
                id=":r3:-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r3:"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Password
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-tddfjc"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-kcn0n4-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Sign in
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
